x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_bulk_action_mutation.ts

```diff
@@ -18,40 +18,40 @@ import { DETECTION_ENGINE_RULES_BULK_ACTION } from '../../../../../common/consta
 export const BULK_ACTION_MUTATION_KEY = ['POST', DETECTION_ENGINE_RULES_BULK_ACTION];
 
 export const useBulkActionMutation = (
-  options?: UseMutationOptions<BulkActionResponse, Error, BulkActionProps>
+  options?: UseMutationOptions<BulkActionResponse, Error, BulkAction>
 ) => {
   const invalidateFindRulesQuery = useInvalidateFindRulesQuery();
   const invalidateFetchRuleByIdQuery = useInvalidateFetchRuleByIdQuery();
   const invalidateFetchTagsQuery = useInvalidateFetchTagsQuery();
   const invalidateFetchPrebuiltRulesStatusQuery = useInvalidateFetchPrebuiltRulesStatusQuery();
   const updateRulesCache = useUpdateRulesCache();
 
-  return useMutation<BulkActionResponse, Error, BulkActionProps>(
-    (action: BulkActionProps) => performBulkAction(action),
+  return useMutation<BulkActionResponse, Error, BulkAction>(
+    (bulkActionDescriptor: BulkAction) => performBulkAction(bulkActionDescriptor),
     {
       ...options,
       mutationKey: BULK_ACTION_MUTATION_KEY,
       onSuccess: (...args) => {
-        const [res, { action }] = args;
-        switch (action) {
-          case BulkAction.enable:
-          case BulkAction.disable: {
+        const [res, { type: actionType }] = args;
+        switch (actionType) {
+          case BulkActionType.enable:
+          case BulkActionType.disable: {
```

Comment:
FWIW I'm fine with this specific renaming, `BulkActionType` is a more precise name.